R/serialized_tests.R

```diff
@@ -158,27 +164,41 @@ style_empty <- function(text) {
 #' @describeIn test_transformer Transformations for indention based on curly
 #'   brackets only.
 style_indent_curly <- function(text) {
+
+  transformers <- list(
+    filler     = create_filler,
+    line_break = NULL,
+    space      =  partial(indent_curly, indent_by = 2),
+    token      = NULL,
+    eol        = strip_eol_spaces,
+    NULL
+  )
+
   text %>%
     compute_parse_data_nested() %>%
-    pre_visit(funs = c(create_filler,
-                   partial(indent_curly, indent_by = 2),
-                   strip_eol_spaces)) %>%
-
+    apply_transformers(transformers) %>%
     serialize_parse_data_nested()
 }
 
 
 #' @describeIn test_transformer Transformations for indention based on curly
 #'   brackets and round brackets.
 style_indent_curly_round <- function(text) {
+  transformers <- list(
+    filler     = create_filler,
+    line_break = NULL,
+    space      =  c(partial(indent_curly, indent_by = 2),
+                    partial(indent_round, indent_by = 2)),
+    token      = NULL,
+    eol        = strip_eol_spaces,
+    NULL
+  )
+
   text %>%
     compute_parse_data_nested() %>%
-    pre_visit(funs = c(create_filler,
-                   partial(indent_curly, indent_by = 2),
-                   partial(indent_round, indent_by = 2),
-                   strip_eol_spaces)) %>%
-
+    apply_transformers(transformers) %>%
     serialize_parse_data_nested()
+
 }
 
 #' @describeIn test_transformer Transformations for indention based on operators
```

Comment:
What do you mean regarding `stle_op()`? No, we don't have a function, but I can create one. But I am not sure whether it would make stuff more readable or just pack too much into one function.

```r
make_transformer_nested <- function(transformers) {
  function(text) {
    text <- gsub(" +$", "", text)
    text <- gsub("\t", "        ", text)
    if (is.null(transformers$space)) return(text)

    pd_nested <- compute_parse_data_nested(text)
    transformed_pd <- apply_transformers(pd_nested, transformers)
    # TODO verify_roundtrip
    new_text <- serialize_parse_data_nested(transformed_pd)
    new_text
  }
}
```
would become
```
make_transformer_nested <- function(transformers) {
  function(text) {
    text <- gsub(" +$", "", text)
    text <- gsub("\t", "        ", text)
    if (is.null(transformers$space)) return(text)
    new_text <- parse_transform_serialize(text, transformers)
    new_text
  }
}
```

Anyways, I think the two `gsub()` expressions should be packed in a function since they are rather low-level. What do you think?